### DISCUSSION TITLE
How to Enable autoApprove and Automerge renovate PR in github private repositories

### DISCUSSION THREAD

**Discussion by chandapukiran** [renovatebot/renovate#25886] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### What is your question?

Hi, i have gone through the document on AutoMerge and AutoApprove. I would like to implement it on github private repositories.

I would like to get some idea on how I can implement it.  I am playing around with my own github repository with self-hosted renovate, what i have noticed is that if I don't have the option enabled "branch protection" and "required number of approval before merging" then renovate is able to automerge the PR into the main branch, however when the branch protection is enabled and it requires at least one approvals before merging then renovate can only submit PR. 

I have tried providing github app all the required permission as suggested here : https://github.com/orgs/community/discussions/13836 but still not able to autoMerge the PR.

I also like to understand more on AutoApprove feature, does it auto approves the PR when the PR requires at least one approval. 

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by HonkingGoose** [comment]
`autoApprove=true` only works on the Azure and _GitLab_ (not GitHub!) platforms. So setting `autoApprove=true` does nothing for you on GitHub.

You can use branch protection rules on GitHub with Renovate, but you'll have to change the GitHub settings, or use a bot for approval. Read the [Renovate docs, Required Pull Request reviews](https://docs.renovatebot.com/key-concepts/automerge/#required-pull-request-reviews) to learn more.

Remember to select _at least one_ status check, if you're using branch protection rules. Quote from the `automerge` docs:

> **Automerge and GitHub branch protection rules**
> 
> You must select at least one status check in the _Require status checks to pass before merging_ section of your branch protection rules on GitHub, if you match all three conditions:
> 
> - `automerge=true`
> - `platformAutomerge=true`, Renovate defaults to `true`
> - You use use GitHub's _Require status checks to pass before merging_ branch protection rule
> 
> If you don't select any status check, and you use platform automerge, then GitHub might automerge PRs with failing tests!

**Comment by rarkins** [comment, marked as answer]
You'll need to run one or more of your own Approve bots and also give them appropriate permissions: https://github.com/renovatebot/renovate-approve-bot

GitHub users and bots are not able to self-approve

**Reply by HonkingGoose** [reply]
Hmm, if GitHub users and bots are not able to self-approve, do we need to change the key concept, automerge docs? Here's the relevant section:

> ### Required Pull Request reviews
> 
> If you have mandatory Pull Request reviews then it means Renovate can't automerge its own PR until such a review has happened.
> 
> If you're on `github.com` or GitHub Enterprise Server (`>=3.4`) you can let Renovate bypass the mandatory Pull Request reviews using the "[Allow specified actors to bypass required pull requests](https://github.blog/changelog/2021-11-19-allow-bypassing-required-pull-requests/)" option in your branch protection rules.
> 
> Alternatively, if you use the Mend Renovate App, you can also install the helper apps [renovate-approve](https://github.com/apps/renovate-approve) and [renovate-approve-2](https://github.com/apps/renovate-approve-2) and they will mark all automerging Pull Requests by Renovate as approved.
> These approval helper apps are only available for GitHub.

**Reply by rarkins** [reply]
Actually that text still applies

**Reply by HonkingGoose** [reply]
I don't understand what you're referring to when you say "Actually that text still applies"... Do you mean your first comment is _correct_, and the docs are thus _wrong_. **Or** that the docs are _correct_.

Choose your own adventure time again! What should I do?

1. Do nothing.
2. Drop the "If you're on `github.com` ..." section, because you can't allow Renovate to bypass it's own PR review anyways.
3. Something else, please explain more.

**Reply by rarkins** [reply]
The docs are correct. Users can choose to either:
- Allow Renovate to bypass mandatory reviews, or
- Add automatic approving, using Renovate Approve Bot or some other means
